core/src/main/scala/nl/vroste/zio/kinesis/client/Producer.scala

```diff
@@ -104,16 +132,86 @@ final case class ProducerSettings(
   failedDelay: Duration = 100.millis,
   metricsInterval: Duration = 30.seconds,
   updateShardInterval: Duration = 30.seconds,
-  aggregate: Boolean = false,
-  allowedErrorRate: Double = 0.05,
-  shardPredictionParallelism: Int = 8
+  shardPrediction: Producer.ShardPrediction = Producer.ShardPrediction.Enabled(),
+  aggregation: Producer.Aggregation = Producer.Aggregation.Disabled,
+  batchDuration: Option[Duration] = None
 ) {
-  require(allowedErrorRate > 0 && allowedErrorRate <= 1.0, "allowedErrorRate must be between 0 and 1 (inclusive)")
+  aggregation match {
```

Comment:
Not a fan of this check being necessary, but not sure how to structure it to avoid it without making it too complicated.